FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -572,59 +556,431 @@ The kernel will be configured as follows:
+                                                        MISRA exception.
+                                                        Avoiding casts between
+                                                        pointers and integers is
+                                                        not practical.  Size
+                                                        differences accounted
+                                                        for using
+                                                        portPOINTER_SIZE_TYPE
+                                                        type.  Checked by
+                                                        assert(). */
+
+      /* Check the alignment of the calculated top of stack is correct. */
+      configASSERT((((portPOINTER_SIZE_TYPE)pxTopOfStack &
+                     (portPOINTER_SIZE_TYPE)portBYTE_ALIGNMENT_MASK) == 0UL));
+
+#if (configRECORD_STACK_HIGH_ADDRESS == 1)
+      {
+        /* Also record the stack's high address, which may assist
+         * debugging. */
+        pxTCB->pxEndOfStack = pxTopOfStack;
+      }
+#endif /* configRECORD_STACK_HIGH_ADDRESS */
+    }
+#else  /* portSTACK_GROWTH */
+    {
+      pxTopOfStack = pxTCB->pxStack;
+
+      /* Check the alignment of the stack buffer is correct. */
+      configASSERT((((portPOINTER_SIZE_TYPE)pxTCB->pxStack &
+                     (portPOINTER_SIZE_TYPE)portBYTE_ALIGNMENT_MASK) == 0UL));
+
+      /* The other extreme of the stack space is required if stack checking is
+       * performed. */
+      pxTCB->pxEndOfStack = pxTCB->pxStack + (ulStackDepth - (uint32_t)1);
     }
+#endif /* portSTACK_GROWTH */
+  }
+
+  (void)pxTopOfStack;
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *       if( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE )
+ *            {
+ *                pxTaskStatus->eCurrentState = eRunning;
+ *            }
+ *            ...
+ * @endcode
+ *
+ * Cover the case where xFreeStackSpace is pdTRUE, avoiding the free
+ * stack space query.
+ */
+void test_coverage_vTaskGetInfo_get_free_stack_space(void) {
+  TCB_t xTaskTCBs[1U] = {NULL};
+  TaskStatus_t pxTaskStatus;
+  BaseType_t xFreeStackSpace = pdTRUE;
+  eTaskState taskState = eReady;
+
+  xTaskTCBs[0].uxPriority = 1;
+  xTaskTCBs[0].xTaskRunState = 0;
+  test_prvInitialiseStack(&xTaskTCBs[0], configMINIMAL_STACK_SIZE);
+  xYieldPendings[0] = pdFALSE;
+  pxCurrentTCBs[0] = &xTaskTCBs[0];
+
+  uxTopReadyPriority = 1;
+  uxSchedulerSuspended = pdTRUE;
+
+  vTaskGetInfo(&xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.xTaskNumber);
+  TEST_ASSERT_EQUAL(eRunning, pxTaskStatus.eCurrentState);
+  TEST_ASSERT_EQUAL((UBaseType_t)1, pxTaskStatus.uxCurrentPriority);
+  TEST_ASSERT_EQUAL((UBaseType_t)0, pxTaskStatus.uxBasePriority);
+  TEST_ASSERT_EQUAL(69, pxTaskStatus.usStackHighWaterMark);
```

Comment:
How was this value derived for the high water mark?

My only concern here is that is that this assertion might make the test 'fragile' and we end up having to make follow on changes to keep the test passing. Maybe im over thinking this but if we ran on different architectures (intel 32 bit vs 64 bit, etc) this value would be different. I would err on the side of just removing this assertion